client/src/components/VerticalCollapsibleSplit/index.jsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import * as Icon from 'react-feather';
+class VerticalCollapsibleSplit extends React.Component {
+
+  constructor (props) {
+    super(props);
+    this.collapse = this.collapse.bind(this);
+    this.state = { closed: false };
+  }
+
+  collapse () {
+    this.setState({ closed: !this.state.closed });
+    document.querySelectorAll(`[data-name='${this.props.name}']`).forEach(el => el.classList.toggle('closed'));
+  }
+
+  render () {
+    return (
```

Comment:
you can use this before your ```return``` to clear up your references, it also has faster faster lookup when they're local vars:
```
let {
  props,
  state
} = this;
```

Reply:
Hi,
Could you elaborate on this? 
I've just implemented this but this is throwing an es-lint warning as its stating that those variables are unused?

Reply:
```
  render () {
    let {
      props,
      state,
    } = this;

    return (
      <div className={'vertical-split'}>
        <div className='visible-content' data-name={props.name}>
          {props.top}
          <button className='collapse-button' onClick={this.collapse}>
            {state.closed ? <Icon.PlusCircle className='plus-icon' />  : <Icon.MinusCircle /> }
          </button>
        </div>
        <div className='collapse-content' data-name={props.name}>
          {props.bottom}
        </div>
      </div>
    );
  }
```